packages/dds/merge-tree/src/client.ts

```diff
@@ -113,18 +113,17 @@ export class Client {
      * It is used to get the segment group(s) for the previous operations.
      * @param count - The number segment groups to get peek from the tail of the queue. Default 1.
      */
-    public peekPendingSegmentGroups(count: number = 1) {
+    public peekPendingSegmentGroups(count: number = 1): SegmentGroup | SegmentGroup[] | undefined {
+        let node = this._mergeTree.pendingSegments?.last;
         if (count === 1) {
-            return this._mergeTree.pendingSegments?.last();
+            return node?.data;
         }
-        let taken = 0;
-        return this._mergeTree.pendingSegments?.some(() => {
-            if (taken < count) {
-                taken++;
-                return true;
-            }
-            return false;
-        }, true);
+        const taken: SegmentGroup[] = [];
+        while (taken.length < count && node) {
```

Comment:
we should implement this in O(N) since it does look like it's used in potential production codepaths, even if for potentially rare scenarios (group ops--not sure how much people use that). Should just be able to use `.push` and then `.reverse` at the end.

Reply:
group are mainly use for reconnecting, as during rebase things could split.

is the problem that unshift is enumerates all elements, which requires another loop? i think i can even skip the loop by using an initialized array, and a for loop

Reply:
yeah, unshift is O(N) as it moves all of the array elements over. Using an initialized array + for loop is fine too.